src/modules/dataset.ts

```diff
@@ -19,7 +19,7 @@ function updateDataset(oldVnode: VNode, vnode: VNode): void {
 
   for (key in oldDataset) {
     if (!dataset[key]) {
-      if (d) {
+      if (d && d[key]) {
```

Comment:
... but if you look at the other code path with the `removeAttribute`, we don't want to invoke that one if dataset is available (which is for HTMLElement but not for Element). So, I guess you just need to protect the delete invocation only. something like:

```
if (d) {
   if (key in d) {
       delete d[key];
   }
} else {
   ...
}
```